js-exercises/array-utils/arrayUtils.js

```diff
@@ -0,0 +1,57 @@
+function forEach(array, callbackFn) {
+  if(!array.length) throw new TypeError('Passed argument is not an array');
+
+  if(typeof callbackFn !== "function") throw new TypeError('Requires a callback function to execute');
+
+  let arrayLength = array.length;
+  
+  for(let i = 0; i < arrayLength; i++) {
+    if(array[i]) {
+      callbackFn(array[i], i, array)  
+    }
+  }
+
+  return;
+}
+
+function map(array, callbackFn) {
+  let resultingArray = [];
+  if(!array.length) throw new TypeError('Passed argument is not an array');
+
+  if(typeof callbackFn !== "function") throw new TypeError('Requires a callback function to execute');
+
+  let arrayLength = array.length;
+  
+  for(let i = 0; i < arrayLength; i++) {
+    if(array[i]) {
+      resultingArray.push(callbackFn(array[i], i, array));
+    }
+  }
```

Comment:
#2
The last point in [spec](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-array.prototype.map) says:

> The range of elements processed by map is set before the first call to callbackfn. Elements which are appended to the array after the call to map begins will not be visited by callbackfn. If existing elements of the array are changed, their value as passed to callbackfn will be the value at the time map visits them; elements that are deleted after the call to map begins and before being visited are not visited.

- This point is not taken care of.
Use a shallow copy of array to iterate over, for this issue:
```suggestion
  const arrayCopy = array;
  
  for(let i = 0; i < arrayCopy.length; i += 1) {
    if(arrayCopy[i]) {
      resultingArray.push(callbackFn(arrayCopy[i], i, arrayCopy));
    }
  }
```